system/lib/wasmfs/syscalls.cpp

```diff
@@ -371,7 +410,8 @@ long __syscall_fstat64(long fd, long buf) {
   return doStat(openFile.locked().getFile(), buffer);
 }
 
-__wasi_fd_t __syscall_open(long pathname, long flags, ...) {
+static __wasi_fd_t
+doOpen(char* pathname, long flags, mode_t mode, backend_t backend = nullptr) {
```

Comment:
how about defining the null value for `backend_t` next to the definition of that type, say `static const backend_t NullBackend = nullptr`. That would avoid writing `nullptr` everywhere - all those places would need to be updated eventually if we switched to an fd-like approach where `backend_t` is not a pointer.